storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfig.java

```diff
@@ -146,111 +146,111 @@ public final class RemoteLogManagerConfig {
                                   DEFAULT_REMOTE_LOG_METADATA_MANAGER_CONFIG_PREFIX,
                                   new ConfigDef.NonEmptyString(),
                                   MEDIUM,
                                   REMOTE_LOG_METADATA_MANAGER_CONFIG_PREFIX_DOC)
-                  .defineInternal(REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP, STRING,
+                  .define(REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP, STRING,
                                   null,
                                   new ConfigDef.NonEmptyString(),
                                   MEDIUM,
                                   REMOTE_STORAGE_MANAGER_CLASS_NAME_DOC)
-                  .defineInternal(REMOTE_STORAGE_MANAGER_CLASS_PATH_PROP, STRING,
+                  .define(REMOTE_STORAGE_MANAGER_CLASS_PATH_PROP, STRING,
                                   null,
                                   new ConfigDef.NonEmptyString(),
                                   MEDIUM,
                                   REMOTE_STORAGE_MANAGER_CLASS_PATH_DOC)
-                  .defineInternal(REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP,
+                  .define(REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP,
                                   STRING,
                                   DEFAULT_REMOTE_LOG_METADATA_MANAGER_CLASS_NAME,
                                   new ConfigDef.NonEmptyString(),
                                   MEDIUM,
                                   REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_DOC)
-                  .defineInternal(REMOTE_LOG_METADATA_MANAGER_CLASS_PATH_PROP,
+                  .define(REMOTE_LOG_METADATA_MANAGER_CLASS_PATH_PROP,
                                   STRING,
                                   null,
                                   new ConfigDef.NonEmptyString(),
                                   MEDIUM,
                                   REMOTE_LOG_METADATA_MANAGER_CLASS_PATH_DOC)
-                  .defineInternal(REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_PROP, STRING,
+                  .define(REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_PROP, STRING,
                                   null,
                                   new ConfigDef.NonEmptyString(),
                                   MEDIUM,
                                   REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_DOC)
-                  .defineInternal(REMOTE_LOG_METADATA_CUSTOM_METADATA_MAX_BYTES_PROP,
+                  .define(REMOTE_LOG_METADATA_CUSTOM_METADATA_MAX_BYTES_PROP,
                                   INT,
                                   DEFAULT_REMOTE_LOG_METADATA_CUSTOM_METADATA_MAX_BYTES,
                                   atLeast(0),
                                   LOW,
                                   REMOTE_LOG_METADATA_CUSTOM_METADATA_MAX_BYTES_DOC)
-                  .defineInternal(REMOTE_LOG_INDEX_FILE_CACHE_TOTAL_SIZE_BYTES_PROP,
+                  .define(REMOTE_LOG_INDEX_FILE_CACHE_TOTAL_SIZE_BYTES_PROP,
```

Comment:
This is unused so far until https://issues.apache.org/jira/browse/KAFKA-14912 completes. Hence, this should be internal. Can you please verify the same for rest of the configs as well?

Reply:
Thank you for the feedback @divijvaidya. I wasn't able to generate docs for internal configs due to `Internal configuration won't show up in the docs and aren't intended for general use`. I can address the other comments and mark this PR as draft until KAFKA-14912 is merged, what do you think?

Reply:
> I wasn't able to generate docs for internal configs

which is perfectly fine. We don't want to publish docs for internal configs since they are not being used.

Reply:
ok, since the intention of KAFKA-15294 is to make configs public which we are not ready to do yet, I think we could mark this PR draft not to get merged.

Reply:
Let me clarify. We do indeed want to make the configs public but we only want to make them externally visible if they are implement. Since https://issues.apache.org/jira/browse/KAFKA-14912 is not implemented yet, we can keep it as internal but still have rest of the configs as public.

This will ensure that 3.6 contains public documentation for things which are being released.

Reply:
Ah, understood. I thought the issue is about making the internal configs public and doc them. Since some of the key configs such as enabling remote storage functionality is missing, I thought it would look good odd.